zabbix/stackstate_checks/zabbix/zabbix.py

```diff
@@ -288,25 +288,34 @@ def retrieve_problems(self, url, auth):
             acknowledged = item.get("acknowledged", None)
             # Object id is in case of object=0 a trigger.
             trigger_id = item.get("objectid", None)
-            # for Zabbix versions <4.0 we need to get the trigger.priority
-            severity = item.get("severity", self.get_trigger_priority(url, auth, trigger_id))
+
+            # for Zabbix versions <4.0 we need to get the trigger.priority and if priority doesn't exist then
+            # either trigger is disabled or doesn't exist
+            priority = self.get_trigger_priority(url, auth, trigger_id)
+            severity = item.get("severity", priority)
 
             zabbix_problem = ZabbixProblem(event_id, acknowledged, trigger_id, severity)
 
             self.log.debug("Parsed ZabbixProblem %s." % zabbix_problem)
             if not event_id or not trigger_id or not severity:
                 self.log.warn("Incomplete ZabbixProblem, got: %s" % zabbix_problem)
 
-            yield zabbix_problem
+            if priority is not None:
+                yield zabbix_problem
 
     def get_trigger_priority(self, url, auth, trigger_id):
+        # `monitored` flag make sure we only get enabled triggers
         params = {
             "output": ["priority"],
-            "triggerids": [trigger_id]
+            "triggerids": [trigger_id],
+            "monitored": True
```

Comment:
Unit test? for disabled and enabled triggers.

Reply:
test case added.